scalanet/discovery/src/io/iohk/scalanet/discovery/ethereum/v4/DiscoveryService.scala

```diff
@@ -228,7 +229,13 @@ object DiscoveryService {
       )
     }
 
-    def removePeer(peerId: Node.Id): State[A] =
+    def removePeer(peerId: Node.Id, toAddress: Node.Address => A): State[A] = {
+      // Find any Peer records that correspond to this ID.
+      val peers =
+        nodeMap.get(peerId).map(node => Peer(node.id, toAddress(node.address))).toSeq ++
+          lastPongTimestampMap.keys.filter(_.id == peerId).toSeq ++
```

Comment:
seems wasteful that we are itereating `lastPongTimestampMap` and `bondingResultsMap` maps two times here when finding all records of peer, and later when fitering this peer out. From what i known this collections can be fairly  large, maybe we could do it in one pass ?

Reply:
Sure, I can change that. But this method won't be actually called, I just kept it for backwards compatibility with KRouter.

Reply:
Done.